### PR TITLE
[release-2.3.0 <  T168-MAGE]  Memgraph Magic Functions

### PR DESCRIPTION
**Pull request by jmatak**

### CODE REVIEW

File: docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md [hunk 1]
```diff
@@ -129,6 +129,55 @@ CALL example.procedure(“single argument”) YIELD result;
 Naturally, you may pass in different arguments.
 
 
+## class mgp.function(func: Callable[[…]])
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md:132]: We left out the docs of the python API in Memgraph, but @jmatak wrote this by hand, thus we can include this. I will open a PR in Memgraph repo to have the same docs in `mgp.py`.

File: docs/reference-guide/query-modules/implement-custom-query-modules/custom-query-module-example.md [hunk 2]
```diff
@@ -203,12 +203,66 @@ def write_procedure(context: mgp.ProcCtx,
...
+    if opt_argument is not None:
+      return_arguments.append(opt_argument)
+
+    # Note that we do not need to specify the result Record as long as it is
+    # Memgraph defined value type.
+    return return_arguments
+```
+
+On the first glance, there is huge similarity between defining a function and a
+procedure. Let's talk differences. First difference is the context type.
```
Comment (jbajic) [inline docs/reference-guide/query-modules/implement-custom-query-modules/custom-query-module-example.md:236]:
```suggestion
procedure. Let's talk about differences. The first difference is the context type.
```

**Review by jbajic — APPROVED** [review]
Small suggestions

File: docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md (hunk 1 above)
Comment (antaljanosbenjamin) [reply docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md:132]: [PR](https://github.com/memgraph/memgraph/pull/391/files) for Memgraph.

File: docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md [hunk 3]
```diff
@@ -129,6 +129,55 @@ CALL example.procedure(“single argument”) YIELD result;
...
+
+Register func as a Memgraph function in the current module.
```
[the remark this answers is not in the document]
Comment (antaljanosbenjamin) [reply docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md:134]: For `read_proc` and `write_proc` the docs are generated, however they don't have these ticks. That's why I didn't added them, to be consistent with the generated docs.

File: docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md [hunk 4]
```diff
@@ -463,6 +512,17 @@ Raise InvalidContextError if context is invalid.
...
 ### must_abort()
 
+## class mgp.FuncCtx(graph)
+Bases: `object`
+
+Context of a function being executed.
+
+Access to a FuncCtx is only valid during a single execution of a function in a
+query. You should not globally store a FuncCtx instance. The graph object within
+the FuncCtx is not mutable.
```
[the remark this answers is not in the document]
Comment (antaljanosbenjamin) [reply docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md:521]: The ticks are not present for `ProcCtx`, so I (Matak) didn't add them here to be consistent with the generated docs.

File: docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md [hunk 5]
```diff
@@ -129,6 +129,55 @@ CALL example.procedure(“single argument”) YIELD result;
 Naturally, you may pass in different arguments.
 
 
+## class mgp.function(func: Callable[[…]])
+
+Register func as a Memgraph function in the current module.
+
+`function` is meant to be used as a decorator function to register module
+functions. The registered func needs to be a callable which optionally takes
```
[the remark this answers is not in the document]
Comment (antaljanosbenjamin) [reply docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md:137]: The same sentence is used for `read_proc` and `write_proc`, thus I would like to keep this as is for now. On the PR in Memgraph repo I will update them.

**Review by g-despot — APPROVED** [review]

**Pull request merged**